lib/http.go

```diff
@@ -11,13 +12,15 @@ import (
 type APIClient struct {
 	BaseURL string
 	APIKey  string
+	Ctx     context.Context
 }
 
 // NewAPIClient creates a new instance of APIClient.
-func NewAPIClient(baseURL, apiKey string) *APIClient {
+func NewAPIClient(baseURL, apiKey string, ctx context.Context) *APIClient {
 	return &APIClient{
 		BaseURL: baseURL,
 		APIKey:  apiKey,
+		Ctx:     ctx,
```

Comment:
there's no point exporting these values, as you don't want them changed/accessed directly anyways